geopmdpy/geopmdpy/system_files.py

```diff
@@ -986,32 +986,37 @@ def _load_session_file(self, sess_path):
             self._update_session_file(client_pid)
         else:
            os.rename(sess_path, renamed_path)
-
-def _get_names(exec_name, quiet=True):
-    try:
-        stdout_encoding = sys.stdout.encoding
-        stderr_encoding = sys.stderr.encoding
-    except AttributeError:
-        stdout_encoding = None
-        stderr_encoding = None
-    if stdout_encoding is None:
-        stdout_encoding = locale.getpreferredencoding()
-    if stderr_encoding is None:
-        stderr_encoding = locale.getpreferredencoding()
-
-    try:
-        stderr_encoding = sys.stderr.encoding
-    except AttributeError:
-        stderr_encoding = None
-    if stderr_encoding is None:
-        stderr_encoding = locale.getpreferredencoding()
-    signals_pid = subprocess.Popen(exec_name,
-                                   stdout=subprocess.PIPE,
-                                   stderr=subprocess.PIPE)
-    stdout_bytes, stderr_bytes = signals_pid.communicate()
-    if not quiet:
-        sys.stderr.write(stderr_bytes.decode(encoding=stderr_encoding))
-    return stdout_bytes.decode(encoding=stdout_encoding).split()
+def _get_names():
+    marker = '\n\nBREAK\n\n'
+    pipe_r, pipe_w = os.pipe()
+    pid = os.fork()
+    if pid != 0:
+        os.close(pipe_w)
+        with os.fdopen(pipe_r) as pipe_ro:
+            buffer = pipe_ro.read()
+        _, exit_status = os.waitpid(pid, 0)
```

Comment:
I'm not sure what situation would cause a hang, but more importantly, I'm not sure what sort of time out would be appropriate.  We know this can be pretty slow when reading from the service with levelzero enabled.